File: problems/b/boatparts/boatparts.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

def main():
    p, n = [int(val) for val in input().split()]

    list_boat_parts = []
    for _ in range(n):
        list_boat_parts.append(input())

    end_day = 0
    set_boat_parts = set()
    for day, part in enumerate(list_boat_parts):
        set_boat_parts.add(part)

        if len(set_boat_parts) == p:
            end_day = day + 1
            break

    if end_day == 0:
        print("paradox avoided")
    else:
        print(end_day)

if __name__ == '__main__':
    main()

File: problems/b/boatparts/boatparts.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <bits/stdc++.h>

using namespace std;

int main() {
    int p, n, end_day;

    cin >> p >> n;

    end_day = 0;
    set<string, greater<string> > list_boat_parts;
    for (int i=0; i<n; i++) {
        string boat_part;
        cin >> boat_part;
        list_boat_parts.insert(boat_part);

        if (p == (int)list_boat_parts.size()) {
            end_day = i + 1;
            break;
        }
    }

    if (end_day == 0) {
        cout << "paradox avoided" << endl;
    }
    else {
        cout << end_day << endl;
    }
}